src/config.ts

```diff
@@ -32,6 +32,20 @@ type DeepPartial<T> = {
 
 export type EnvVarReplacements = Record<string, string | number | true>;
 
+/** Snowpack export map */
+export type SnowpackBuildMap = {[outputLoc: string]: SnowpackFile};
+
+/** Standard file interface */
+export interface SnowpackFile {
+  /** base extension (e.g. `.js`) */
+  baseExt: string;
+  /** file contents */
+  code: string;
+  /** expanded extension (e.g. `.proxy.js` or `.module.css`) */
+  expandedExt: string;
+  /** if no location on disk, assume this exists in memory */
+  locOnDisk?: string;
+}
```

Comment:
This is the heart of the change right here: a standard file interface.

It’s tempting to add in something here like `outputLoc` or some key to tell each file what it’s being built _to_. But I think that’s unknowable, and stretches this too far out. What it’s being built to should be the responsibility of the build command, as well as Snowpack plugins. If that output is changing as it’s passing hands, that means we can’t trust it at any given moment, so we shouldn’t even bother.

Instead, we have a thing called a `SnowpackBuildMap` which is just a map of output locations which map to files. That can be added to, or overwritten, and multiple outputs can even point to the same files (which may be helpful for things like `.vue`, which can take a single source and output a `.js` and `.css` file, for example).

All of that concern is beyond this PR; this is just some of the thinking behind this structure.

Reply:
And yes, this will probably become a giant object over time as we need more file info and that’s OK. At least it’s standardized.

I originally thought of having a `SnowpackModule` interface, too, that maps import specifiers back to files on disk… but I’ll have to come back to that one as we get further.

Reply:
+1 for everything outlined above, appreciate how well thought out this is. My only comment is a nit: `SnowpackFile` feels too general, especially when you've got a general concept like this but do want some constraints like not wanting to mix with the `outputLoc`. 

Maybe `SnowpackBuildFile`?  `SnowpackFileOnDisk`? Not picky